src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ISZMetricsCollectorSpark.java

```diff
@@ -0,0 +1,150 @@
+        Long pairsCount = 0L;
+        Double totalLength = 0.0;
+        Double totalSquaredLenth = 0.0;
+        for(Map.Entry<Integer, Long> pair : hist.entrySet()) {
+            final Integer size = pair.getKey();
+            final Long cnt = pair.getValue();
+            pairsCount += cnt;
+            totalLength += ((double) size) * ((double) cnt);
+            totalSquaredLenth += ((double) size) * ((double) size) * ((double) cnt);
+        }
+        metric.READ_PAIRS = pairsCount;
+        final double n = (double) pairsCount;
+        metric.MEAN_INSERT_SIZE = totalLength/n;
+        metric.STANDARD_DEVIATION = Math.sqrt(totalSquaredLenth/n -
+                metric.MEAN_INSERT_SIZE*metric.MEAN_INSERT_SIZE);
+        metric.STANDARD_DEVIATION *= Math.sqrt(n/(n-1));
+
+        // second round get median done
+        final Long midPoint = pairsCount/2;
+        pairsCount = 0L; // reset
+        for(Map.Entry<Integer, Long> pair : hist.entrySet()){
+            final Integer size = pair.getKey();
+            final Long cnt = pair.getValue();
+            pairsCount += cnt;
+            if(pairsCount > midPoint) {
+                metric.MEDIAN_INSERT_SIZE = size;
+                break;
+            }else if(pairsCount.equals(midPoint)){ // borderline case
+                final Integer nextSize = hist.higherKey(size);
+                metric.MEDIAN_INSERT_SIZE = ((double)(size + nextSize))/2.0;
+                break;
+            }
+        }
+
+        // third round to get MAD done (is there any functional way/map to do this?)
+        TreeMap<Double, Long> medianDiff = new TreeMap<Double, Long>();
+        for(Map.Entry<Integer, Long> pair : hist.entrySet()){
+            Integer size = pair.getKey();
+            Long cnt = pair.getValue();
+            double diff =  Math.abs((double)size - metric.MEDIAN_INSERT_SIZE);
+            if(medianDiff.get(diff)!=null){
+                medianDiff.put(diff, medianDiff.get(diff)+cnt);
+            }else{
+                medianDiff.put(diff, cnt);
+            }
+        }
+        pairsCount = 0L;
+        for(Map.Entry<Double, Long> pair : medianDiff.entrySet()){
+            final Double diff = pair.getKey();
+            final Long cnt = pair.getValue();
+            pairsCount += cnt;
+            if(pairsCount > midPoint) {
+                metric.MEDIAN_ABSOLUTE_DEVIATION = diff;
+                break;
+            }else if(pairsCount.equals(midPoint)){ // borderline case
+                final double nextDiff = medianDiff.higherKey(diff);
+                metric.MEDIAN_ABSOLUTE_DEVIATION = ((diff + nextDiff))/2.0;
+                break;
+            }
+        }
+    }
+
+    private void collectPercentages(InsertSizeMetrics metric){
+
+        double bins[][] = new double[10][2];
+        final double offsets[] = {0.005, 0.05, 0.10, 0.15, 0.2, 0.25, 0.30, 0.35, 0.40, 0.45};
+        // final round, percentiles
+        for(int i=0; i<10; ++i){
```

Comment:
Can this whole thing be replaced by using `Percentile` from Apache (use setData to put the data once and then `evaluate` as needed)? That applies to the median computation too. (btw maybe you can then also eliminate the sorting on your end because Percentile does the sorting - actually it does not even sort the whole array but only sorts what it needs).